### PR TITLE
chore: Fix CocoaPods install on Linux

### PR DESCRIPTION
**Pull request by legobeat**
## **Description**

Fixes and enables CocoaPods installation on Linux. After this change, macOS is not required to update `Podfile.lock`.

- Enable `pod install` step in `yarn setup` script on all platforms, not just macOS
  - Failing installation is still only treated as actual failure on macOS
-  Apply `react-native` patch that resolves installation error due to attempt at patching missing Xcode
  - Upstream PR: https://github.com/facebook/react-native/pull/44417
- Add developer `Dockerfile` which can be used locally to run routine development tasks like `yarn setup`. 
  - Add CI job that builds docker image and tests it with metamask-mobile repo 
    - In order for the new job to run, the newly used actions must be enabled in repository settings. [Preview run](https://github.com/legobeat/metamask-mobile/actions/runs/9292287827/job/25572841157)

## **Related issues**

- #9231

## **Manual testing steps**

1. Follow steps in `scripts/docker/README.md` to build and run docker image
2. In container from built image, run:  `yarn setup`
3. Inspect that local pod installation completes

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by legobeat** [comment]
This has only been tested on `amd64` so far. Would be great if someone on `aarch64` (like Apple Silicon) could test it as well.

File: patches/react-native+0.71.15.patch [hunk 1]
```diff
@@ -237,4 +237,19 @@ index 290bd23..20d85e0 100644
...
+--- a/node_modules/react-native/scripts/react_native_pods.rb
++++ b/node_modules/react-native/scripts/react_native_pods.rb
+@@ -224,7 +224,9 @@ def react_native_post_install(installer, react_native_path = "../node_modules/re
+   ReactNativePodsUtils.exclude_i386_architecture_while_using_hermes(installer)
+   ReactNativePodsUtils.fix_library_search_paths(installer)
+   ReactNativePodsUtils.set_node_modules_user_settings(installer, react_native_path)
+-  ReactNativePodsUtils.apply_xcode_15_patch(installer)
++  if Environment.new().ruby_platform().include?('darwin')
++    ReactNativePodsUtils.apply_xcode_15_patch(installer)
++  end
```
Comment (leotm) [inline patches/react-native+0.71.15.patch:252]: added 🍒⛏️ cherry-pick request request https://github.com/reactwg/react-native-releases/issues/296
to remove this patch after upgrading to RN 0.74.x

**Review by Cal-L — APPROVED** [review]
LGTM

**Comment by legobeat** [comment]
- Enable `pod install` step in `yarn setup` script on all platforms
- Failing installation is still only treated as actual failure on macOS
- Apply `react-native` patch that resolves installation error due to
attempt at patching missing Xcode
  - Upstream PR: https://github.com/facebook/react-native/pull/44417
- Add developer `Dockerfile` which can be used locally to run routine
development tasks like `yarn setup`.
- Add CI job that builds docker image and tests it with metamask-mobile
- In order for the new job to run, the newly used actions must be
enabled in repository settings.

**Pull request merged**